cosmovisor/scanner_test.go

```diff
@@ -0,0 +1,87 @@
+package cosmovisor
+
+import (
+	"bufio"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitForInfo(t *testing.T) {
+	cases := map[string]struct {
+		write         []string
+		expectUpgrade *UpgradeInfo
+		expectErr     bool
+	}{
+		"no match": {
+			write: []string{"some", "random\ninfo\n"},
+		},
+		"match height with no info": {
+			write: []string{"first line\n", `UPGRADE "myname" NEEDED at height: 123: `, "\nnext line\n"},
```

Comment:
These test against some hardcoded strings.

Since we now are in the same repo, can we just refer to the `x/upgrade` package and use a function there to generate the log message (the same function it uses to print)?

Reply:
I am facing a problem with that, since we use a different go.mod when I get cosmos-sdk I got:

```
go: finding module for package google.golang.org/grpc/naming
../../../../go/pkg/mod/google.golang.org/api@v0.13.0/internal/pool.go:10:2: module google.golang.org/grpc@latest found (v1.31.0), but does not contain package google.golang.org/grpc/naming

Compilation finished with exit code 1

```